openmetadata-ui/src/main/resources/ui/src/pages/DatasetDetailsPage/DatasetDetailsPage.component.tsx

```diff
@@ -250,60 +208,16 @@ const DatasetDetailsPage: FunctionComponent = () => {
     }
   };
 
-  const fetchTabSpecificData = async (tabField = '') => {
-    switch (tabField) {
-      case TabSpecificField.SAMPLE_DATA: {
-        if (!isUndefined(sampleData)) {
-          break;
-        } else {
-          setIsSampleDataLoading(true);
-
-          try {
-            const res = await getTableDetailsByFQN(tableFQN, tabField);
-            const { sampleData } = res;
-            setSampleData(sampleData as TableData);
-          } catch (error) {
-            showErrorToast(
-              error as AxiosError,
-              t('server.entity-details-fetch-error', {
-                entityType: t('label.table'),
-                entityName: datasetFQN,
-              })
-            );
-          } finally {
-            setIsSampleDataLoading(false);
-          }
-
-          break;
-        }
-      }
-
-      case TabSpecificField.ACTIVITY_FEED: {
-        getFeedData();
-
-        break;
-      }
-
-      default:
-        break;
-    }
-  };
-
   useEffect(() => {
-    if (datasetTableTabs[activeTab - 1].path !== tab) {
-      setActiveTab(getCurrentDatasetTab(tab));
+    if (EntityTabs.ACTIVITY_FEED === tab) {
+      getFeedData();
     }
-    setEntityThread([]);
-  }, [tab]);
-
-  useEffect(() => {
-    fetchTabSpecificData(datasetTableTabs[activeTab - 1].field);
-  }, [activeTab, feedCount]);
+  }, [tab, feedCount]);
```

Comment:
On tab change, we should reset the entity thread state here, also why we need `feedCount` as a dependency here?

Reply:
can you check again, you are mixing to useEffect changes, here and github is creating some weird diff, I have removed on useEffect which was not necessary as we are accessing tab change via useParam, and 2nd useEffect is for calling `fetchTabSpecificData` which was performing one action based on condition so i have remove that function and added that to useEffect. but github has created some weird diff mixing changes of both useEffect

Reply:
No worries.